### ISSUE TITLE
Improvements to spin buttons

### ISSUE THREAD

**Issue by acburdine** [TryGhost/Ghost#5650]
With the merge of #5633, a spin-button component was added to the ember app, and it was placed into most places in which it was needed.

However, there are a couple of improvements that can be made to some existing places and some new places that the component can be added to.

- [x] Fix spinner disappearing too quickly on signin

When signing in to the Ghost blog, the spinner disappears after pressing "Sign In", but before the page changes, so there is a bit of a delay between the two. There are a couple of ways I can think of/have tried to fix this. I tried adding the code to change the button from spinner back to button into the `sessionAuthenticationSucceeded` method of the application route. This decreased the delay somewhat, but it was still noticeable. The only other way I could think of to do this is to add something into the authenticated route file that would reset the spinner button on the signin controller. The only problem with this is that it would apply to every route and it would run regardless of whether or not signin just happened, which seemed to me to be overkill.

- [ ] Add spin buttons to modals

While adding a spin-button to the login modal was fairly simple, adding it to other modals presents somewhat of a complex problem. Due to how the modal submit/close actions work, it makes managing the spinner through async requests tricky. This would be nice to have for certain things like the importer.

**Comment by ErisDS** [comment]
Spinner was fixed by #5789 / 4b6f28d

**Comment by ErisDS** [comment]
As the remaining part of this issue relates to modals, and is tricky because of their current implementation, I think this can be closed in favour of #5798 - once the refactor is done this issue can be opened again
